exasol_advanced_analytics_framework/query_handler/context/top_level_query_handler_context.py

```diff
@@ -163,13 +184,20 @@ def _invalidate(self):
         self._valid_object_proxies = set()
         self._owned_object_proxies = set()
         self._is_valid = False
-        child_context_were_not_released = False
+        not_released_child_contexts = []
+        exceptions_from_not_released_child_contexts = []
         for child_query_handler_context in self._child_query_handler_context_list:
             if child_query_handler_context._is_valid:
-                child_context_were_not_released = True
-                child_query_handler_context._invalidate()
-        if child_context_were_not_released:
-            raise RuntimeError("Child contexts are not released.")
+                not_released_child_contexts.append(child_query_handler_context)
+                try:
+                    child_query_handler_context._invalidate()
+                except ChildContextNotReleasedError as e:
+                    exceptions_from_not_released_child_contexts.append(e)
```

Comment:
Okay, now I get what this actually means. How about renaming this to `exceptions_thrown_by_not_released_child_context`?

Because I completely misinterpreted the word `exceptions` in the context before "exception from" in my eyes suggests "this, but not that".

Reply:
yeah, this is probably a better name, will change it